ansible/files/admin_api_scripts/pg_egress_collect.pl

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env perl
+
+# This script receive tcpdump output through STDIN and does:
+#
+# 1. extract outgoing TCP packet length on all devices port 5432 and 6543
+# 2. sum the length up to one minute
+# 3. save the total length to file (default is /tmp/pg_egress_collect.txt) per minute
+#
+# Usage:
+#
+# tcpdump -s 128 -Q out -i any -nn -tt -vv -p -l 'tcp and (port 5432 or port 6543)' | perl pg_egress_collect.pl /tmp/output.txt
+#
+
+use POSIX;
+use List::Util qw(sum);
+use IO::Async::Loop;
+use IO::Async::Stream;
+use IO::Async::Timer::Periodic;
+
+use strict;
+use warnings;
+
+# total captured packets lenth in a time frame
+my $captured_len = 0;
+
+# extract tcp packet length captured by tcpdump
+#
+# Sample input lines:
+#
+# 1674013833.940253 IP (tos 0x0, ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 60)
+#     10.112.101.122.5432 > 220.235.16.223.62599: Flags [S.], cksum 0x5de3 (incorrect -> 0x63da), seq 2314200657, ack 2071735457, win 62643, options [mss 8961,sackOK,TS val 3358598837 ecr 1277499190,nop,wscale 7], length 0
+# 1674013833.989257 IP (tos 0x0, ttl 64, id 24975, offset 0, flags [DF], proto TCP (6), length 52)
+#     10.112.101.122.5432 > 220.235.16.223.62599: Flags [.], cksum 0x5ddb (incorrect -> 0xa25b), seq 1, ack 9, win 490, options [nop,nop,TS val 3358598885 ecr 1277499232], length 0
+sub extract_packet_length {
+    my ($line) = @_;
+
+    #print("debug: >> " . $line);
+
+    if ($line =~ /^(\d+)\.\d+/) {
+        # extract packet timestamp, do nothing for now
```

Comment:
not familiar with perl -- when do we do something with the timestamp?

Reply:
that was supposed to be used in the future if we need timestamp, but for now it seems not need, will remove it.